rfcs/0122-system-permission.md

```diff
@@ -0,0 +1,712 @@
+# Motivation
+
+The `Administrator` role currently combines both project administration functionality and system administration functionality. Giving a user access to do things like delete a signal or create a shared API key cannot be done without also giving them access to do things like installing apps and modifying the Seq license. 
+
+The Seq permissions system currently has four permissions, Read, Write, Setup, and Ingest. These are allocated to four built in roles, Administrator, User (read-only), User (read/write) and User (read/write/ingest).
+
+The Administrator role currently encompasses both project administration, and host system access, through functionality such as Seq App installation.
+
+This RFC proposes splitting the Setup permission into Project (project administration) and System (host system administration), so that these two responsibilities can be separated. Team leaders often need access to project administration features such as API keys and retention policies, but don't necessarily need privileged access to the host system.
+
+The built in Administrator role will include both Project and System, i.e., retaining the functionality of this role today. A new role Project Owner will have Project permissions, but not System.
+
+# Detailed Design
+
+## Roles to Permission Mapping
+
+| Role | 2021.4 permissions | 2022.1 permissions |
+| -- | -- | -- |
+| User (read-only) | Read | Read |
+| User (read/write) | Read, Write | Read, Write |
+| User (read/write/ingest) | Read, Write, Ingest | Read, Write, Ingest |
+| Project Owner | | Read, Write, Ingest, Project |
+| Administrator | Read, Write, Ingest, Setup | Read, Write, Ingest, Project, System |
+
+## 2021.4 Permission to API Endpoint Mapping
+
+### `api/` 
```

Comment:
Perhaps we could narrow this list down to only show endpoints affected by the change, and use diff syntax to highlight what's changing?

```diff
- | `api/alertstate/` | `GET` | `Setup` | | |
+ | `api/alertstate/` | `GET` | `Project` | | |
```

Reply:
Diff syntax breaks the table formatting and unformatted markdown tables are hard to read. Would you prefer tables or diff syntax?

Reply:
I'm good with whatever illustrates it best - just throwing ideas around :-)